flink-yarn/src/main/java/org/apache/flink/yarn/YarnResourceManager.java

```diff
@@ -395,10 +405,13 @@ public void onNodesUpdated(List<NodeReport> list) {
 
 	@Override
 	public void onError(Throwable error) {
-		onFatalError(error);
+		runAsync(() -> onFatalError(error));
```

Comment:
I think it would be good to let the error propagate directly. In case of an OOM exception we want to quickly shut down the JVM.

Reply:
done